net/procfs/net_udp.c

```diff
@@ -0,0 +1,157 @@
+ * "License"); you may not use this file except in compliance with the
+ * License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <nuttx/config.h>
+
+#include <sys/types.h>
+#include <stdio.h>
+#include <string.h>
+#include <debug.h>
+
+#include <arpa/inet.h>
+#include <netinet/in.h>
+
+#include <nuttx/net/netstats.h>
+
+#include "procfs/procfs.h"
+#include "udp/udp.h"
+
+/****************************************************************************
+ * Pre-processor Definitions
+ ****************************************************************************/
+
+#if defined(CONFIG_NET_UDP) && defined(CONFIG_NET_IPv4)
+
+#define UDP_LINELEN 120
+
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: netprocfs_read_udpstats
+ *
+ * Description:
+ *   Read and format TCP statistics.
+ *
+ * Input Parameters:
+ *   priv - A reference to the network procfs file structure
+ *   buffer - The user-provided buffer into which network status will be
+ *            returned.
+ *   bulen  - The size in bytes of the user provided buffer.
+ *
+ * Returned Value:
+ *   Zero (OK) is returned on success; a negated errno value is returned
+ *   on failure.
+ *
+ ****************************************************************************/
+
+ssize_t netprocfs_read_udpstats(FAR struct netprocfs_file_s *priv,
+                                FAR char *buffer, size_t buflen)
+{
+  FAR struct udp_conn_s *conn = NULL;
+  char remote[INET_ADDRSTRLEN + 1];
+  char local[INET_ADDRSTRLEN + 1];
+  struct in_addr laddr;
+  struct in_addr raddr;
+  int skip = 1;
+  int len = 0;
+
+  net_lock();
+
+  if (udp_nextconn(NULL) == NULL)
+    {
+      goto end_lock;
+    }
+
+  if (priv->offset == 0)
+    {
+      len = snprintf(buffer, buflen, "UDP sl     "
+                                     "local_address          "
+                                     "remote_address    "
+                                     " flg   "
+#if CONFIG_NET_SEND_BUFSIZE > 0
+                                     "txsz   "
+#endif
+                                     "rxsz"
+#ifdef CONFIG_NET_IPv6
+                                     "  (IPv4)"
```

Comment:
remove, let's support both ipv4 and ipv6

Reply:
Done